src/lib.rs

```diff
@@ -1,7 +1,317 @@
+use std::{collections::HashMap};
+
+#[derive(Debug)]
+pub enum Action<'a> {
+    Unknown(String),
+    Incorrect(String, &'a Verb<'a>),
+    BadParameter(String, &'a Command),
+    Run(Vec<ParameterValue<'a>>),
+    Help(Vec<ParameterValue<'a>>),
+    Exit,
+}
+
+pub trait Informational {
+    fn get_help(&self) -> &Manual;
+}
+
+#[derive(Debug)]
+pub struct Manual<'a> {
+    short_description: &'a str,
+    detailed_help: Vec<&'a str>,
+}
+
+impl<'a> Manual<'a> {
+    pub fn new(short_description: &'a str, detailed_help: Vec<&'a str>) -> Manual<'a> {
+        Manual {
+            short_description,
+            detailed_help,
+        }
+    }
+}
+
+pub struct Runtime {}
+
+pub struct Parser<'a> {
+    verbs: HashMap<String, Verb<'a>>,
+}
+
+impl<'a> Parser<'a> {
+    pub fn new(verbs: Vec<Verb>) -> Parser {
+        let mut verbs_map = HashMap::with_capacity(verbs.len());
+
+        for verb in verbs {
+            verbs_map.insert(verb.name().clone(), verb);
+        }
+
+        Parser {
+            verbs: verbs_map,
+        }
+    }
+
+    pub fn parse(&self, input: String) -> Action {
+        let (verb_name, remaining_commands, matching_verb) = match input.split_once(" ") {
+            Some((verb_name, remaining_commands)) => {
+                (verb_name.to_owned(), remaining_commands, self.verbs.get(verb_name))
+            },
+            None => {
+                (input.clone(), "", self.verbs.get(&input))
+            },
+        };
+
+        let action = match matching_verb {
+            Some(verb) => {
+                verb.parse(remaining_commands)
+            },
+            None => Action::Unknown(verb_name),
+        };
+
+        action
+    }
+}
+
+#[derive(Debug)]
+pub struct Verb<'a> {
+    name: String,
+    verbs: HashMap<String, Verb<'a>>,
+    commands: HashMap<String, Command>,
+    manual: Manual<'a>,
+}
+
+impl<'a> Verb<'a> {
+    pub fn new(name: &str, verbs: Option<Vec<Verb<'a>>>, commands: Option<Vec<Command>>, manual: Manual<'a>) -> Verb<'a> {
+        let mut verbs_map = HashMap::new();
+
+        if let Some(verbs) = verbs {
+            for verb in verbs {
+                verbs_map.insert(verb.name().clone(), verb);
+            }
+        }
```

Comment:
Do you need an `Option<Vec<...>>` here?  Dropping the `Option` will allow you to drop the `if let Some(...)` calls if you allow a consumer to pass in an empty `Vec`.